pkg/crd/testdata/cronjob_types.go

```diff
@@ -133,6 +135,39 @@ type CronJobSpec struct {
 	// +kubebuilder:default={}
 	DefaultedEmptyObject EmpiableObject `json:"defaultedEmptyObject"`
 
+	// This tests that kubebuilder defaulting takes precedence.
+	// +kubebuilder:default="kubebuilder-default"
+	// +default="kubernetes-default"
+	DoubleDefaultedString string `json:"doubleDefaultedString"`
+
+	// This tests that primitive defaulting can be performed.
+	// +default="forty-two"
+	KubernetesDefaultedString string `json:"kubernetesDefaultedString"`
+
+	// This tests that slice defaulting can be performed.
+	// +default=["a","b"]
+	KubernetesDefaultedSlice []string `json:"kubernetesDefaultedSlice"`
+
+	// This tests that object defaulting can be performed.
+	// +default=[{"nested": {"foo": "baz", "bar": true}},{"nested": {"foo": "qux", "bar": false}}]
+	KubernetesDefaultedObject []RootObject `json:"kubernetesDefaultedObject"`
```

Comment:
This technically tests defaulting of a slice :)

Reply:
heh, copied from `DefaultedObject []RootObject `json:"defaultedObject"`` above ... I guess it defaults a slice of objects, so... both... but sure :)

Reply:
do you want me to change this and the existing one above, or leave them, or just change the new one?

Reply:
Maybe let's change both. I guess this works anyway, but maybe safer to default an object to get the actual coverage for this case